src/net/if_.rs

```diff
@@ -0,0 +1,48 @@
+//! Network interface name resolution.
+//!
+//! Uses Linux and/or POSIX functions to resolve interface names like "eth0"
+//! or "socan1" into device numbers.
+
+use libc::{c_char, c_uint};
+use std::ffi::{CString, NulError};
+use std::io;
+
+/// Error that can occur during interface name resolution.
+#[derive(Debug)]
+pub enum NameToIndexError {
+    /// Failed to allocate a C-style string to for the syscall
+    NulError,
+    IOError(io::Error),
+}
+
+impl From<NulError> for NameToIndexError {
+    fn from(_: NulError) -> NameToIndexError {
+        NameToIndexError::NulError
+    }
+}
+
+impl From<io::Error> for NameToIndexError {
+    fn from(e: io::Error) -> NameToIndexError {
+        NameToIndexError::IOError(e)
+    }
+}
+
+extern {
+    fn if_nametoindex(ifname: *const c_char) -> c_uint;
+}
+
+/// Resolve an interface into a interface number.
+pub fn name_to_index(name: &str) -> Result<c_uint, NameToIndexError> {
```

Comment:
A few things:

* use the functions name as in C
* return a `nix::Result<c_uint>`, which is equivalent to swapping in `nix::Error` for the error type
* use a [`NixPath`](http://rustdoc.s3-website-us-east-1.amazonaws.com/nix/master/linux/nix/trait.NixPath.html) bound for the string argument (the name is bad, and it'll likely change soon, see #221 if you want more info!)

This would look like:

``` rust
pub fn if_nametoindex<S: ?Sized + NixPath>(target: &P) -> Result<c_uint> {
    let res = try!(name.with_nix_path(|name| { /* rest of body here */ }));

    from_ffi(res)
}
```

The [`nix::from_ffi()`](http://rustdoc.s3-website-us-east-1.amazonaws.com/nix/master/linux/nix/fn.from_ffi.html) function handles errno translation for you. :-)